src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs

```diff
@@ -339,15 +429,30 @@ private void ImportCall(ILOpcode opcode, int token)
                 }
             }
 
-            if (method.OwningType.IsDelegate && method.Name == "Invoke" &&
-                opcode != ILOpcode.ldftn && opcode != ILOpcode.ldvirtftn)
+
+            // TODO-LLVM: delete when IL->LLVM module is gone
+            if (_compilation.TargetArchIsWasm())
             {
-                // This call is expanded as an intrinsic; it's not an actual function call.
-                // Before codegen realizes this is an intrinsic, it might still ask questions about
-                // the vtable of this virtual method, so let's make sure it's marked in the scanner's
-                // dependency graph.
-                _dependencies.Add(_factory.VTable(method.OwningType), reason);
-                return;
+                if (!method.IsRuntimeDeterminedExactMethod && method.IsVirtual && !method.HasInstantiation &&
+                    MetadataVirtualMethodAlgorithm.FindSlotDefiningMethodForVirtualMethod(method) == method)
+                {
+                    _dependencies.Add(_factory.VirtualMethodUse(method), "LLVM delegate Invoke");
+                }
+            }
```

Comment:
Is this needed?

Reply:
Is this only needed for delegates? I'm asking because the dependency says `"LLVM delegate Invoke"`, but it's not nested under the IsDelegate check.

Reply:
I think I'll just have to change the reason for this dependency.   I tried moving inside the IsDelegate check and the scanner was ok with that, but it crashes at runtime

```
RuntimeError: function signature mismatch
    at S_P_Reflection_Core_System_Collections_Concurrent_ConcurrentUnifierW_2_Container<System___Canon__System___Canon>__TryGetValue (wasm-function[11850]:0xafe8b5)
    at S_P_Reflection_Core_System_Collections_Concurrent_ConcurrentUnifierW_2<System___Canon__System___Canon>__GetOrAdd (wasm-function[11849]:0xafd07a)
    at S_P_Reflection_Core_System_Reflection_Runtime_Dispensers_DispenserThatReusesAsLongAsValueIsAlive_2<System___Canon__System___Canon>__GetOrAdd (wasm-function[18710]:0xdc2f86)
    at S_P_Reflection_Core_System_Reflection_Runtime_Assemblies_RuntimeAssembly__TryGetRuntimeAssembly (wasm-function[9950]:0x8df653)
    at S_P_Reflection_Core_System_Reflection_Runtime_Assemblies_RuntimeAssembly__GetRuntimeAssembly (wasm-function[9949]:0x8df383)
    at S_P_Reflection_Core_System_Reflection_Runtime_TypeInfos_NativeFormat_NativeFormatRuntimeNamedTypeInfo__get_Assembly (wasm-function[11909]:0xb0d704)
    at S_P_CoreLib_System_Text_StringBuilder__AppendSpanFormattable<Int32> (wasm-function[7537]:0x5e96f6)
    at S_P_CoreLib_System_Text_StringBuilder__Append_10 (wasm-function[15580]:0xd2b34a)
    at S_P_CoreLib_System_Reflection_AssemblyNameFormatter__ComputeDisplayName (wasm-function[4480]:0x3b72af)
    at S_P_CoreLib_System_Reflection_RuntimeAssemblyName__get_FullName (wasm-function[4478]:0x3b648a)
    at S_P_Reflection_Core_System_Reflection_Runtime_TypeInfos_NativeFormat_NativeFormatRuntimeNamedTypeInfo__get_InternalFullNameOfAssembly (wasm-function[11897]:0xb0a3e8)
    at S_P_Reflection_Core_System_Reflection_Runtime_TypeInfos_RuntimeTypeInfo__get_AssemblyQualifiedName (wasm-function[10919]:0x9e71e4)
    at S_P_Reflection_Core_System_Reflection_Runtime_TypeInfos_RuntimeConstructedGenericTypeInfo__get_FullName (wasm-function[12033]:0xb23aae)
    at HelloWasm_Program__TestMetaData (wasm-function[5813]:0x4cb693)
    at HelloWasm_Program__Main (wasm-function[5877]:0x4da893)
    at HelloWasm__Module___MainMethodWrapper (wasm-function[16292]:0xd4c0a7)
    at StartupCodeMain (wasm-function[7327]:0x595598)
```

Reply:
Changed the reason, not great, but its "temporary"...